Terminal.Gui/Views/TableView/ListTableSource.cs

```diff
@@ -0,0 +1,228 @@
+				table.Columns.Add (new DataColumn (col.ToString ()));
+			}
+			for (int row = 0; row < (Count / table.Columns.Count); row++) {
+				table.Rows.Add ();
+			}
+			// return partial row
+			if (Count % table.Columns.Count != 0) {
+				table.Rows.Add ();
+			}
+
+			return table;
+		}
+
+		private DataTable TransposeTable (DataTable initialTable, bool includeColumnNames = false)
+		{
+			var newTable = new DataTable ();
+			
+			int offset = 0;
+			if (includeColumnNames) {
+				// This creates a first column to contain existing column names
+				// (TODO: Allow formatting such a column like a header)
+				newTable.Columns.Add (new DataColumn ("0"));
+				offset++;
+			}
+
+			for (int i = 0; i < initialTable.Columns.Count; i++) {
+				DataRow row = newTable.NewRow ();
+				if (includeColumnNames) {
+					row [0] = initialTable.Columns [i].ColumnName;
+				}
+				for (int j = offset; j < initialTable.Rows.Count + offset; j++) {
+					if (newTable.Columns.Count < initialTable.Rows.Count + offset)
+						newTable.Columns.Add (new DataColumn (j.ToString ()));
+					row [j] = initialTable.Rows [j - offset] [i];
+				}
+				newTable.Rows.Add (row);
+			}
+
+			return newTable;
+		}
+
+		/// <summary>
+		/// Returns the size in characters of the longest value read from <see cref="ListTableSource.List"/>
+		/// </summary>
+		/// <returns></returns>
+		private int CalculateMaxLength ()
+		{
+			if (List == null || Count == 0) {
+				return 0;
+			}
+
+			int maxLength = 0;
+			foreach (var t in List) {
+				int l;
+				if (t is ustring u) {
+					l = TextFormatter.GetTextWidth (u);
+				} else if (t is string s) {
+					l = s.Length;
+				} else {
+					l = t.ToString ().Length;
+				}
+
+				if (l > maxLength) {
+					maxLength = l;
+				}
+			}
+
+			return maxLength;
+		}
+
+		private int CalculateColumns ()
+		{
+			int cols;
+
+			int colWidth = CalculateMaxLength ();
+			if (colWidth > tableView.MaxCellWidth) {
+				colWidth = tableView.MaxCellWidth;
+			}
+
+			if (tableView.MinCellWidth > 0 && colWidth < tableView.MinCellWidth) {
+				if (tableView.MinCellWidth > tableView.MaxCellWidth) {
+					colWidth = tableView.MaxCellWidth;
+				} else {
+					colWidth = tableView.MinCellWidth;
+				}
+			}
+			if (Style.VerticalOrientation != Style.ScrollParallel) {
+				float f = (float)tableView.Bounds.Height - tableView.GetHeaderHeight ();
+				cols = (int)Math.Ceiling (Count / f);
+			} else {
+				cols = ((int)Math.Ceiling (((float)tableView.Bounds.Width - 1) / colWidth)) - 2;
+			}
+
+			return (cols > 1) ? cols : 1;
+		}
+
+		/// <summary>
+		/// Defines rendering options that affect how the view is displayed.
+		/// </summary>
+		public class ListColumnStyle {
+
+			/// <summary>
+			/// Gets or sets a flag indicating whether to populate data down each column rather than across each row.
+			/// Defaults to <see langword="false"/>.
+			/// </summary>
+			public bool VerticalOrientation { get; set; } = false;
```

Comment:
It would be consistent with the rest of the library if this was `public Orientation Orientation { get ; set; } = Orientation.Horizontal`.

Not sure if it's a big enough deal to change tho.

Reply:
I agree that `Orientation` is clearer.  You write an API only once and then have to maintain it indefinitely so its worth getting it right if possible please.

Maybe?

```csharp
/// <summary>
/// Gets or sets the direction list elements are laid out before wrapping
/// to the next row/column.
/// </summary>
/// <remarks>
/// With <see cref="Orientation.Horizontal"/> (the default) each element
/// is positioned in a new cell on the same row until the screen edge is
/// reached.  With <see cref="Orientation.Vertical"/>, elements are positioned
/// vertically in the first column until the bottom of the screen is reached.
/// </remarks>
public Orientation LayoutOrientation { get; set; } = Orientation.Horizontal;
```

Reply:
Thanks.  I'll switch to Orientation, but @tznind FYI your description is assuming ScrollParallel is false.